File: src/main/java/edu/cornell/cs/apl/viaduct/util/dataflow/IdentityEdge.java
package edu.cornell.cs.apl.viaduct.util.dataflow;

import org.jgrapht.graph.DefaultEdge;

/** An edge that passes values through unmodified. */
public class IdentityEdge<A> extends DefaultEdge implements DataFlowEdge<A> {
  @Override
  public final A propagate(A in) {
    return in;
  }

  @Override
  public String toString() {
    return "";
  }
}

File: compiler/src/main/kotlin/io/github/aplcornell/viaduct/util/dataflow/IdentityEdge.kt
package io.github.aplcornell.viaduct.util.dataflow

import org.jgrapht.graph.DefaultEdge

/** An edge that passes values through unmodified. */
class IdentityEdge<A> : DefaultEdge(), DataFlowEdge<A> {
    override fun propagate(input: A): A {
        return input
    }

    override fun toString(): String {
        return ""
    }
}
